tests/trinity/core/p2p-proto/test_sync.py

```diff
@@ -157,72 +155,33 @@ def finalizer():
 
 
 @pytest.fixture
-def chaindb_20():
-    chain = PoWMiningChain.from_genesis(FakeAsyncAtomicDB(), GENESIS_PARAMS, GENESIS_STATE)
-    for i in range(20):
-        tx = chain.create_unsigned_transaction(
-            nonce=i,
-            gas_price=1234,
-            gas=1234000,
-            to=RECEIVER.public_key.to_canonical_address(),
-            value=i,
-            data=b'',
-        )
-        chain.apply_transaction(tx.as_signed_transaction(SENDER))
-        chain.mine_block()
+def leveldb_20(tmpdir):
+    fresh_path = tmpdir.join('20-headers.ldb')
```

Comment:
Somewhat trivial but the stdlib versions have context manager APIs for temporary directories have context manager APIs which do cleanup for you.